Sonobuoy running multiple plugins failing with nil pointer exception

**What steps did you take and what happened:**
sonobuoy run --plugin <plugin-yaml-path1> --plugin <plugin-yaml-path2> is failing with the below error:
PS D:\azure-arc-kubernetes-conformance\src\plugins> sonobuoy run --plugin .\core\conformance.yaml --plugin 
.\arc_agent_cleanup\arc_agent_cleanup.yaml

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x38 pc=0x179b1a2]

goroutine 1 [running]:
github.com/vmware-tanzu/sonobuoy/pkg/client.autoAttachResultsDir(0xc00028df10, 0x2, 0x2, 0x1bd7275, 0x15)
        /home/runner/work/sonobuoy/sonobuoy/pkg/client/gen.go:295 +0x62
github.com/vmware-tanzu/sonobuoy/pkg/client.(*SonobuoyClient).GenerateManifestAndPlugins(0xc000399830, 0xc00030a180, 0x3, 0x28a1550, 0xc000778160, 0xc00080fd30, 0x5fe8ff, 0x38, 0x2839c70, 0x1bc8a01)
        /home/runner/work/sonobuoy/sonobuoy/pkg/client/gen.go:204 +0xcb7
github.com/vmware-tanzu/sonobuoy/pkg/client.(*SonobuoyClient).GenerateManifest(...)
        /home/runner/work/sonobuoy/sonobuoy/pkg/client/gen.go:89
github.com/vmware-tanzu/sonobuoy/pkg/client.(*SonobuoyClient).Run(0xc000399830, 0xc00030a180, 0x28a1550, 0x0)
        /home/runner/work/sonobuoy/sonobuoy/pkg/client/run.go:212 +0x11e
github.com/vmware-tanzu/sonobuoy/cmd/sonobuoy/app.submitSonobuoyRun.func1(0xc00012f400, 0xc000093480, 0x0, 0x4)
        /home/runner/work/sonobuoy/sonobuoy/cmd/sonobuoy/app/run.go:100 +0x24a
github.com/spf13/cobra.(*Command).execute(0xc00012f400, 0xc000093400, 0x4, 0x4, 0xc00012f400, 0xc000093400)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000dac80, 0x0, 0x1906640, 0xc00007e058)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        /home/runner/work/sonobuoy/sonobuoy/main.go:27 +0x2e 

**What did you expect to happen:**
Successfully able to run both plugins

**Anything else you would like to add:**

**Environment:**
- Sonobuoy version: v0.55.0 
- Kubernetes version: (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.9", GitCommit:"ed93e7ea8f833824d05578765c2ad3efbebbfebe", GitTreeState:"clean", BuildDate:"2021-10-21T03:31:08Z", GoVersion:"go1.15.14", Compiler:"gc", Platform:"linux/amd64"}
- Kubernetes installer & version: AKS
- Cloud provider or hardware configuration: Azure 
- OS (e.g. from `/etc/os-release`): 
- Sonobuoy tarball (which contains * below)

--- a/pkg/client/gen.go
+++ b/pkg/client/gen.go
@@ -291,11 +291,11 @@ func applyEnvOverrides(pluginEnvOverrides map[string]map[string]string, plugins 
 // autoAttachResultsDir will either add the volumemount for the results dir or modify the existing
 // one to have the right path set.
 func autoAttachResultsDir(plugins []*manifest.Manifest, resultsDir string) {
-	for i := range plugins {
-		containers := []*corev1.Container{&plugins[i].Spec.Container}
-		if plugins[i].PodSpec != nil {
-			for i := range plugins[i].PodSpec.Containers {
-				containers = append(containers, &plugins[i].PodSpec.Containers[i])
+	for pluginIndex := range plugins {
+		containers := []*corev1.Container{&plugins[pluginIndex].Spec.Container}
+		if plugins[pluginIndex].PodSpec != nil {
+			for containerIndex := range plugins[pluginIndex].PodSpec.Containers {
+				containers = append(containers, &plugins[pluginIndex].PodSpec.Containers[containerIndex])
 			}
 		}
 		addOrUpdateResultsMount(resultsDir, containers...)

--- a/test/integration/sonobuoy_integration_test.go
+++ b/test/integration/sonobuoy_integration_test.go
@@ -649,6 +649,10 @@ func TestExactOutput_LocalGolden(t *testing.T) {
 			desc:       "warning given if rerun-failed set withother mode/focus/skip flags",
 			cmdLine:    "gen --rerun-failed testdata/results-4-e2e-failures.tar.gz --mode=certified-conformance --kubernetes-version=ignore",
 			expectFile: "testdata/gen-mode-and-rerun.golden",
+		}, {
+			desc:       "multiple plugins and multiple containers issue 1528",
+			cmdLine:    "gen -p testdata/plugins/good/sidecar.yaml -p testdata/plugins/good/hello-world.yaml --kubernetes-version=ignore",
+			expectFile: "testdata/gen-issue-1528.golden",
 		},
 	}
 	for _, tc := range testCases {

--- a/test/integration/testdata/gen-issue-1528.golden
+++ b/test/integration/testdata/gen-issue-1528.golden
@@ -1,0 +1,215 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: sonobuoy
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    component: sonobuoy
+  name: sonobuoy-serviceaccount
+  namespace: sonobuoy
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    component: sonobuoy
+    namespace: sonobuoy
+  name: sonobuoy-serviceaccount-sonobuoy
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: sonobuoy-serviceaccount-sonobuoy
+subjects:
+- kind: ServiceAccount
+  name: sonobuoy-serviceaccount
+  namespace: sonobuoy
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    component: sonobuoy
+    namespace: sonobuoy
+  name: sonobuoy-serviceaccount-sonobuoy
+rules:
+- apiGroups:
+  - '*'
+  resources:
+  - '*'
+  verbs:
+  - '*'
+- nonResourceURLs:
+  - '/metrics'
+  - '/logs'
+  - '/logs/*'
+  verbs:
+  - 'get'
+---
+apiVersion: v1
+data:
+  config.json: |
+    {"Description":"DEFAULT","UUID":"","Version":"*STATIC_FOR_TESTING*","ResultsDir":"/tmp/sonobuoy/results","Resources":null,"Filters":{"Namespaces":".*","LabelSelector":""},"Limits":{"PodLogs":{"Namespaces":"kube-system","SonobuoyNamespace":true,"FieldSelectors":[],"LabelSelector":"","Previous":false,"SinceSeconds":null,"SinceTime":null,"Timestamps":false,"TailLines":null,"LimitBytes":null}},"QPS":30,"Burst":50,"Server":{"bindaddress":"0.0.0.0","bindport":8080,"advertiseaddress":"","timeoutseconds":21600},"Plugins":null,"PluginSearchPath":["./plugins.d","/etc/sonobuoy/plugins.d","~/sonobuoy/plugins.d"],"Namespace":"sonobuoy","WorkerImage":"sonobuoy/sonobuoy:*STATIC_FOR_TESTING*","ImagePullPolicy":"IfNotPresent","ImagePullSecrets":"","AggregatorPermissions":"clusterAdmin","ProgressUpdatesPort":"8099","SecurityContextMode":"nonroot"}
+kind: ConfigMap
+metadata:
+  labels:
+    component: sonobuoy
+  name: sonobuoy-config-cm
+  namespace: sonobuoy
+---
+apiVersion: v1
+data:
+  plugin-0.yaml: |
+    sonobuoy-config:
+      description: This is a plugin description.
+      driver: Job
+      plugin-name: hello-world
+      result-format: raw
+      source-url: foo.com
+    spec:
+      command:
+      - ./run.sh
+      env:
+      - name: RESULTS_DIR
+        value: /tmp/sonobuoy/results
+      - name: SONOBUOY
+        value: "true"
+      - name: SONOBUOY_CONFIG_DIR
+        value: /tmp/sonobuoy/config
+      - name: SONOBUOY_K8S_VERSION
+        value: ignore
+      - name: SONOBUOY_PROGRESS_PORT
+        value: "8099"
+      - name: SONOBUOY_RESULTS_DIR
+        value: /tmp/sonobuoy/results
+      image: hello:v9
+      imagePullPolicy: IfNotPresent
+      name: plugin
+      resources: {}
+      volumeMounts:
+      - mountPath: /tmp/sonobuoy/results
+        name: results
+  plugin-1.yaml: |
+    podSpec:
+      containers:
+      - env:
+        - name: RESULTS_DIR
+          value: /tmp/sonobuoy/results
+        - name: SONOBUOY
+          value: "true"
+        - name: SONOBUOY_CONFIG_DIR
+          value: /tmp/sonobuoy/config
+        - name: SONOBUOY_K8S_VERSION
+          value: ignore
+        - name: SONOBUOY_PROGRESS_PORT
+          value: "8099"
+        - name: SONOBUOY_RESULTS_DIR
+          value: /tmp/sonobuoy/results
+        image: foo/baz:0.1.2
+        imagePullPolicy: Always
+        name: sidecar
+        resources: {}
+        volumeMounts:
+        - mountPath: /tmp/sonobuoy/results
+          name: results
+      restartPolicy: Never
+      serviceAccountName: sonobuoy-serviceaccount
+    sonobuoy-config:
+      driver: Job
+      plugin-name: sidecarplugin
+      result-format: junit
+    spec:
+      env:
+      - name: RESULTS_DIR
+        value: /tmp/sonobuoy/results
+      - name: SONOBUOY
+        value: "true"
+      - name: SONOBUOY_CONFIG_DIR
+        value: /tmp/sonobuoy/config
+      - name: SONOBUOY_K8S_VERSION
+        value: ignore
+      - name: SONOBUOY_PROGRESS_PORT
+        value: "8099"
+      - name: SONOBUOY_RESULTS_DIR
+        value: /tmp/sonobuoy/results
+      image: foo/bar:0.1.2
+      imagePullPolicy: IfNotPresent
+      name: plugin
+      resources: {}
+      volumeMounts:
+      - mountPath: /tmp/sonobuoy/results
+        name: results
+kind: ConfigMap
+metadata:
+  labels:
+    component: sonobuoy
+  name: sonobuoy-plugins-cm
+  namespace: sonobuoy
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    component: sonobuoy
+    sonobuoy-component: aggregator
+    tier: analysis
+  name: sonobuoy
+  namespace: sonobuoy
+spec:
+  securityContext:
+    runAsUser: 1000
+    runAsGroup: 3000
+    fsGroup: 2000
+  containers:
+  - env:
+    - name: SONOBUOY_ADVERTISE_IP
+      valueFrom:
+        fieldRef:
+          fieldPath: status.podIP
+    image: sonobuoy/sonobuoy:*STATIC_FOR_TESTING*
+    imagePullPolicy: IfNotPresent
+    name: kube-sonobuoy
+    command: ["/sonobuoy"]
+    args: ["aggregator", "--no-exit", "--level=info", "-v=4", "--alsologtostderr"]
+    volumeMounts:
+    - mountPath: /etc/sonobuoy
+      name: sonobuoy-config-volume
+    - mountPath: /plugins.d
+      name: sonobuoy-plugins-volume
+    - mountPath: /tmp/sonobuoy
+      name: output-volume
+  restartPolicy: Never
+  serviceAccountName: sonobuoy-serviceaccount
+  tolerations:
+  - key: "kubernetes.io/e2e-evict-taint-key"
+    operator: "Exists"
+  volumes:
+  - configMap:
+      name: sonobuoy-config-cm
+    name: sonobuoy-config-volume
+  - configMap:
+      name: sonobuoy-plugins-cm
+    name: sonobuoy-plugins-volume
+  - emptyDir: {}
+    name: output-volume
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    component: sonobuoy
+    sonobuoy-component: aggregator
+  name: sonobuoy-aggregator
+  namespace: sonobuoy
+spec:
+  ports:
+  - port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    sonobuoy-component: aggregator
+  type: ClusterIP
+

--- a/test/integration/testdata/plugin-list.golden
+++ b/test/integration/testdata/plugin-list.golden
@@ -9,3 +9,9 @@ Filename: testdata/plugins/good/hw-2.yaml
 Plugin name (in aggregator): hello-world2
 Source URL: 
 Description: 
+---
+Run as: sidecar
+Filename: testdata/plugins/good/sidecar.yaml
+Plugin name (in aggregator): sidecarplugin
+Source URL: 
+Description: 

--- a/test/integration/testdata/plugins/good/sidecar.yaml
+++ b/test/integration/testdata/plugins/good/sidecar.yaml
@@ -1,0 +1,19 @@
+podSpec:
+  containers:
+    - name: sidecar
+      image: foo/baz:0.1.2
+      imagePullPolicy: Always
+  restartPolicy: Never
+  serviceAccountName: sonobuoy-serviceaccount
+sonobuoy-config:
+  driver: Job
+  plugin-name: sidecarplugin
+  result-format: junit
+spec:
+  image: foo/bar:0.1.2
+  imagePullPolicy: Always
+  name: plugin
+  resources: {}
+  volumes:
+  - name: results
+    emptyDir: {}

